### DISCUSSION TITLE
How do you load a user on the server?

### DISCUSSION THREAD

**Discussion by alexwhitmore** [supabase/supabase#9015] [Questions]
I am attempting to create a login system in the Astro framework.

I am able to register users, login, and log out when everything is loaded **client** side. However, when unauthenticated, navigating to a page I don't have access to still shows a sliver of content which I don't like.

I was thinking that if I load the user on the **server** side, it would load faster and prevent that sliver of unauthenticated content.

In the home screen, I have a form where I am logging in with a user and password, which will then redirect me to a new page, `/dashboard`.
```js
  const handleSignUp = async (e) => {
    e.preventDefault();

    try {
      const { data, error } = await supabase.auth.signInWithPassword({
        email: email,
        password: password,
      });
    } catch (error) {
      if (error instanceof Error) {
        alert(error.message);
      }
    } finally {
      location.href = '/dashboard';
    }
  };
```

And here on the `/dashboard` page:
```jsx
---
import { supabase } from '../utils/supabase';
const {
  data: { user },
} = await supabase.auth.getUser();

console.log(user); // This returns null
---

<script>
  import { supabase } from '../utils/supabase';

  const {
    data: { user },
  } = await supabase.auth.getUser();

  console.log(user); // This successfully loads on the client side
</script>

<section>
  <h1>Dashboard</h1>
  <p>I should only see this page if I am authenticated.</p>
</section>
``` 

In the code block in-between the fences, that JS will run on the server whereas the script tags will run client side. As I mentioned, client side, I can log the user information after a log in no problem.

I see that Supabase has helper functions for Nextjs and Svelte to to do this. Are there different built-in methods to load users on the server? How can I check on the server if this newly registered/logged in user exists??

**Comment by saltcod** [comment]
Hey @alexwhitmore 

We don't have an Astro-specific example, but our auth-helpers examples might give you some pointers: 
https://github.com/supabase/auth-helpers/tree/main/examples
